src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,32 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object rectangle) {
+        if (rectangle == this) {
+            return true;
+        }
+        if (rectangle == null) {
+            return false;
+        }
+        if (rectangle.getClass().equals(Rectangle.class)) {
+            Rectangle rectangle1 = (Rectangle) rectangle;
+            return (this.color == rectangle1.color
+                    || (this.color != null && this.color.equals(rectangle1.color)))
+                    && (this.length == rectangle1.length
+                    || (this.length != null && this.length.equals(rectangle1.length)))
+                    && (this.width == rectangle1.width
+                    || (this.width != null && this.width.equals(rectangle1.width)));
```

Comment:
let's remove this keyword
https://mateacademy.slack.com/archives/G01EHB8CWAE/p1607422728299900